src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskInitiateByIdControllerTest.java

```diff
@@ -96,43 +97,56 @@ public void should_return_a_201_when_initiating_a_task_by_id() {
-            .body(
-                "task_role_resources[0].role_name",
-                anyOf(is("tribunal-caseworker"), is("senior-tribunal-caseworker"))
-            )
-            .body("task_role_resources[0].read", equalTo(true))
-            .body("task_role_resources[0].own", equalTo(true))
-            .body("task_role_resources[0].execute", equalTo(false))
-            .body("task_role_resources[0].cancel", equalTo(true))
-            .body("task_role_resources[0].refer", equalTo(true))
-            .body("task_role_resources[0].authorizations", equalTo(emptyList()))
-            .body("task_role_resources[0].auto_assignable", equalTo(false))
-            .body(
-                "task_role_resources[0].role_category",
-                anyOf(is("LEGAL_OPERATIONS"), is(nullValue()))
-            )
-            .body("task_role_resources[1].task_role_id", notNullValue())
-            .body("task_role_resources[1].task_id", equalTo(taskId))
-            .body(
-                "task_role_resources[1].role_name",
-                anyOf(is("tribunal-caseworker"), is("senior-tribunal-caseworker"))
-            )
-            .body("task_role_resources[1].read", equalTo(true))
-            .body("task_role_resources[1].own", equalTo(true))
-            .body("task_role_resources[1].execute", equalTo(false))
-            .body("task_role_resources[1].cancel", equalTo(true))
-            .body("task_role_resources[1].refer", equalTo(true))
-            .body("task_role_resources[1].authorizations", equalTo(emptyList()))
-            .body("task_role_resources[1].auto_assignable", equalTo(false))
-            .body("task_role_resources[1].role_category",
-                anyOf(is("LEGAL_OPERATIONS"), is(nullValue())))
+            .body("task_role_resources.size()", equalTo(3))
             .body("work_type_resource.id", equalTo("decision_making_work"))
             .body("work_type_resource.label", equalTo("Decision-making work"));
 
+        assertPermissions(
+            getTaskResource(result, "tribunal-caseworker"),
+            Map.of("read", true,
+                   "refer", true,
+                   "own", true,
+                   "manage", true,
+                   "execute", false,
+                   "cancel", true,
+                   "task_id", taskId,
+                   "authorizations", List.of("IA"),
+                   "auto_assignable", false
+            )
+        );
+
+        assertPermissions(
+            getTaskResource(result, "tribunal-caseworker"),
+            Map.of("read", true,
+                   "refer", true,
+                   "own", true,
+                   "manage", true,
+                   "execute", false,
+                   "cancel", true,
+                   "task_id", taskId,
+                   "authorizations", List.of("IA"),
+                   "auto_assignable", false
+            )
+        );
+
+        Optional.ofNullable(getTaskResource(result, "task-supervisor"))
```

Comment:
This is optional so the test will work both ways which is great.  
We might need to double check with the business if it is ok for the `task-supervisor` role to be put onto Tasks in R1.